API/SearchService/src/util.py

```diff
@@ -0,0 +1,73 @@
+from enum import Enum
+import azure.functions as func
+
+from .exception import ParameterException
+
+class SearchType(int, Enum):
+    All = 1
+    Scenario = 2
+    Command = 3
+
+    def get_search_fields(self):
+        if self == self.All:
+            return ["scenario", "description", "commandSet/command"]
+        elif self == self.Scenario:
+            return ["scenario", "description"]
+        elif self == self.Command:
+            return ["commandSet/command"]
+
+
+class RequiredParameter:
+    pass
+
+
+def get_param(req: func.HttpRequest, name: str, default=RequiredParameter):
+    value = req.params.get(name)
+    if not value:
+        try:
+            req_body = req.get_json()
+            value = req_body.get(name, default)
+        except ValueError:
+            pass
+    if value == RequiredParameter:
+        raise ParameterException(f'Illegal parameter: please pass in the parameter "{name}"')
+    return value
+
+
+def get_param_str(req: func.HttpRequest, name: str, default=RequiredParameter):
+    value = get_param(req, name, default)
+    if not isinstance(value, str):
+        raise ParameterException(f'Illegal parameter: the parameter "{name}" must be the type of string')
+    return value
```

Comment:
Is this a redundant method?

Reply:
> Is this a redundant method?

We extract parameters from `req.params` and `req.body`. The request body is a json and `int`/`bool`/`list` can be a member of json.